src/pgrn-custom-scan.c

```diff
@@ -827,6 +924,11 @@ PGrnEndCustomScan(CustomScanState *customScanState)
 		grn_obj_close(ctx, state->searched);
 		state->searched = NULL;
 	}
+	if (state->sorted)
+	{
+		grn_obj_close(ctx, state->sorted);
+		state->sorted = NULL;
+	}
```

Comment:
We should close `sorted` AFTER `searched` because `searched` is referred from `sorted`.